src/components/application_manager/src/message_helper/message_helper.cc

```diff
@@ -1128,6 +1224,65 @@ void MessageHelper::SendAllOnButtonSubscriptionNotificationsForApp(
   }
 }
 
+smart_objects::SmartObjectSPtr
+MessageHelper::CreateOnButtonSubscriptionNotification(
+    uint32_t app_id,
+    hmi_apis::Common_ButtonName::eType button,
+    bool is_subscribed) {
+  using namespace smart_objects;
+  using namespace hmi_apis;
+  LOG4CXX_AUTO_TRACE(logger_);
+  SmartObjectSPtr notification_ptr =
+      std::make_shared<SmartObject>(SmartType_Map);
+  if (!notification_ptr) {
```

Comment:
necessary to make this check? we don't usually check for alloc fails

Reply:
@iCollin removed in https://github.com/smartdevicelink/sdl_core/pull/3475/commits/0149f2ffce888d0985f5329a83791c48e55f9dbd